usb/usb_host_uvc/src/descriptor.c

```diff
@@ -0,0 +1,1030 @@
+/*
+ * SPDX-FileCopyrightText: 2007 Daniel Drake <dsd@gentoo.org>
+ * SPDX-FileCopyrightText: 2001 Johannes Erdfelt <johannes@erdfelt.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
```

Comment:
Can we release it under Apache license with LGPL code inside?

Reply:
I kept original copyright and added SPDX before is as suggested in https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/copyright-guide.html#third-party-licenses.

Reply:
As far as I know, LGPL can be re-licensed as long as user of the finished software can modify, re-compile or replace the portion of the software that is licensed under the LGPL licence.

Reply:
In this case, since we are statically linking LGPL licensed code with (possibly proprietary) application, this wouldn't be possible. So we can't include LGPL code here.

So we'll have to find some way to re-implement this part of the code, or find another implementation under a suitable license (BSD, MIT, Apache).

Reply:
Thakn you @igrr for clarification. Second half of the file is written by me, and the first half deals with turning raw configuration descriptor into libusb aware config descriptor. I'll re-implement it.